### PR TITLE
Using GitHub Actions for Release and PR testing

### PR DESCRIPTION
**Pull request by kddejong**
*Issue #, if available:*

*Description of changes:*
- Converting from Travis to GitHub Actions for PR Testing and release

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .github/workflows/pypi.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        with:
+          python-version: "3.x"
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install setuptools wheel twine
+      - name: Build and publish
+        env:
+          TWINE_USERNAME: ${{ secrets.PYPI_USERNAME }}
+          TWINE_PASSWORD: ${{ secrets.PYPI_PASSWORD }}
```
Comment (PatMyron) [inline .github/workflows/pypi.yaml:23]: [don't see these set yet btw](https://github.com/aws-cloudformation/cfn-python-lint/settings/secrets/actions)

Comment (kddejong) [reply .github/workflows/pypi.yaml:23]: Done

File: .github/workflows/pypi.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
+name: Upload Python Package
+
+on:
+  release:
+    types: [created]
```
Comment (jotompki) [inline .github/workflows/pypi.yaml:5]:
```suggestion
    types: [published]
```

File: .github/workflows/pypi.yaml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        uses: actions/setup-python@v1
+        with:
+          python-version: "3.x"
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install setuptools wheel twine
+      - name: Build and publish
+        env:
+          TWINE_USERNAME: ${{ secrets.PYPI_USERNAME }}
```
Comment (jotompki) [inline .github/workflows/pypi.yaml:22]: https://github.com/marketplace/actions/pypi-publish use the pypi publish action with a scoped token for the cfn-python-lint package. simplifies to where you just need to include the command like here: https://github.com/aws-cloudformation/cloudformation-cli/blob/master/.github/workflows/pypi-release.yaml#L26

File: .github/workflows/pypi.yaml [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python
+        uses: actions/setup-python@v1
+        with:
+          python-version: "3.x"
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
```
Comment (jotompki) [inline .github/workflows/pypi.yaml:18]: i don't think you need this line. python setup action should handle this.

**Review by jotompki — COMMENTED** [review]
Take a look at our release job here:https://github.com/aws-cloudformation/cloudformation-cli/blob/master/.github/workflows/pypi-release.yaml

Looks good just would use a pypi token and use the pypi publish action

**Review by jotompki — APPROVED** [review]

**Comment by jotompki** [comment]
Should we remove travis as well?

**Comment by PatMyron** [comment]
> Should we remove travis as well?

I think this is still a subset of our old Travis tests:

https://github.com/aws-cloudformation/cfn-python-lint/blob/3b677733ef2fc8abc28756f6acd41219d1b0baff/.travis.yml#L30-L33

**Comment by kddejong** [comment]
Converted the integration tests and removed `.travis.yml`

**Review by PatMyron — APPROVED** [review]

File: .github/workflows/test.yaml [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+          file: ./coverage.xml # optional
+          name: py${{ matrix.python-version }}-${{ matrix.env.TOXENV }}-${{ matrix.os }}
+          flags: unittests # optional
+          fail_ci_if_error: true # optional (default = false)
+  integration:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+        python: [3.8]
```
Comment (PatMyron) [inline .github/workflows/test.yaml:37]: could add more Python versions here if you want, but fine either way

**Pull request merged**